bubblesub/cmd/sub_set.py

```diff
@@ -27,7 +28,48 @@ class SubtitlesSetCommand(BaseCommand):
 
     @property
     def is_enabled(self) -> bool:
-        return self.args.target.makes_sense
+        if self.args.target.makes_sense:
+            return True
+
+        if self.api.gui.current_view == ViewState.Full:
+            return True
+
+        if (
+            self.api.gui.current_view == ViewState.Subs
+            and self.args.start is not None
+            and self.args.start.expr == "ps.e"
+        ):
+            return True
+
+        if (
+            self.api.gui.current_view == ViewState.Subs
+            and self.args.end is not None
+            and self.args.end.expr == "ns.s"
+        ):
+            return True
+
+        if (
+            self.api.gui.current_view == ViewState.Audio
+            and self.args.start is not None
+            and self.args.start.expr == "a.s"
+        ):
+            return True
+
+        if (
+            self.api.gui.current_view == ViewState.Video
+            and self.args.start is not None
+            and "f" in self.args.start.expr
+        ):
+            return True
+
+        if (
+            self.api.gui.current_view == ViewState.Video
+            and self.args.end is not None
+            and "f" in self.args.end.expr
+        ):
+            return True
+
+        return False
```

Comment:
No, no, no…
Instead of going after every command, amend subtitle selection parsing, and once it tries to access audio or video APIs, check if they're in the right view, if not, raise `CommandUnavailable`.

Reply:
Completely reverted my changes. Actually, I don't like the `CommandUnavailable` approach because I think it's pretty horrible to notify the user through the console about the impossibility to run a command in a view. So I thought two different approaches:
- the first one consists of splitting the `sub-set` command in three subcommands: `sub-set-audio`, `sub-set-video`, and `sub-set`
- the second one aims to add another parameter to the command called view that specifies in which view the command should run

Let me know what you think about this approaches

Reply:
>Completely reverted my changes. Actually, I don't like the CommandUnavailable approach because I think it's pretty horrible to notify the user through the console about the impossibility to run a command in a view. So I thought two different approaches:

This is what happens already in some edge cases. Did you notice it? I don't think so. So let's go with `CommandUnavailable`.

Reply:
Yes, I have noticed that but it doesn't mean that I like it... 
Anyway, I think I'm not going to change this file, so let's iterate over the `sub-set` command eventually.

Reply:
PTS and SubtitleSelection classes are entirely orthogonal to the commands which is why this is done this way

Reply:
Another point to consider is that PTS for some configurations fires a UI dialog, so if you'd like to poll whether a PTS is available for ex. by calling `@property def is_enabled(self) -> bool: try: … except CommandUnavailable: return False` you'd sometimes accidentally open UI dialogs, which is bad

Reply:
It's a kind of last resort approach for lazily evaluated expressions that might or might not make sense at runtime. So far it wasn't worth the effort to make this better.

Reply:
> Another point to consider is that PTS for some configurations fires a UI dialog, so if you'd like to poll whether a PTS is available for ex. by calling `@property def is_enabled(self) -> bool: try: … except CommandUnavailable: return False` you'd sometimes accidentally open UI dialogs, which is bad

Yep, I've seen those UI dialog methods in the PTS class. I don't have particular problems with the SubtitleSelection class since it acts on subs and doesn't consider streams, so it's okay. 

Anyway, I don't know yet what you think about my two approaches. I mean, are they feasible? Why don't you like them? In this way, I can think a different temporary approach if you're still interested in the `sub-set` command for this PR.

Reply:
I don't like either, because `sub-set` is primarily focused on subtitles and has nothing to do with audio / video unless the user is trying to invoke a pathological example such as `sub-set --start=av.s` (e.g. set subtitle start to audio view start) while the spectrogram is hidden. But the command shouldn't be enabled because it's the PTS value that causes the problem, not the command itself. Also PTS is used in other commands such as `play-region`, `sub-split`, `save-screenshot` etc. etc..

Reply:
> I don't like either, because `sub-set` is primarily focused on subtitles and has nothing to do with audio / video

True

> the user is trying to invoke a pathological example such as `sub-set --start=av.s` (e.g. set subtitle start to audio view start) while the spectrogram is hidden.
But the command shouldn't be enabled because it's the PTS value that causes the problem, not the command itself. Also PTS is used in other commands such as `play-region`, `sub-split`, `save-screenshot` etc. etc..

So the plan would be the following one for now:
1. Raise the `CommandUnavailable` exception in Pts when a particular video/audio API is launched
2. Intercept the exception in `is_enabled` for each command you have specified above
3. Extract the UI dialogs from PTS not to accidentally open them (??)

Reply:
>in order to not accidentally open them (??)

If you don't know what I mean, consider what happens if `--pts=ask-time`, and we try to evaluate it in `is_enabled`. We get a dialog whenever we check if command is available.

But yeah it sounds like a good plan but out of scope for this PR. LMK when it's ready for reveiw

Reply:
> > in order to not accidentally open them (??)
> 
> If you don't know what I mean, consider what happens if `--pts=ask-time`, and we try to evaluate it in `is_enabled`. We get a dialog whenever we check if command is available.
> 
Yep, I've seen the code, my double question marks were referred to the possibility to extract the dialog functions outside PTS in some way.

> But yeah it sounds like a good plan but out of scope for this PR. LMK when it's ready for reveiw

I agree. Let me implement the `console` and `node-editor` toggles first and then you can review it.